### PR TITLE
Own template

### PR DESCRIPTION
**Pull request by niamhdougan**
Adding gitlab ci template

### CODE REVIEW

File: my_template.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  script:
+    - pipenv install --dev
+
+unit_tests_and_coverage:
+  tags:
+    - niamh
+  stage: test
+  coverage: '/TOTAL.*\s+(\d+\%)/'
+  script:
+    - pipenv run tests
```
Comment (thomascobb) [inline my_template.yml:23]: We might as well add back in the coverage html publishing now

File: my_template.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+install_packages:
+  tags:
+    - niamh
+  stage: install
+  script:
+    - pipenv install --dev
+
+unit_tests_and_coverage:
+  tags:
+    - niamh
```
Comment (thomascobb) [inline my_template.yml:19]: What does this do?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -1,20 +1,2 @@
...
-  script:
-    - pipenv run docs
-    - pipenv run tests
-    - dls-python3 setup.py bdist_wheel
-    - dls-py3 check --no-tag-check
-  after_script:
-    - mkdir -p ${DOCS_PUBLISH_ROOT}/latest
-    - cp -rf build/html/* ${DOCS_PUBLISH_ROOT}/latest
-
+  - local: /my_template.yml
```
Comment (thomascobb) [inline .gitlab-ci.yml:2]: I assume my_template will eventually move to the controls common templates...

File: my_template.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    - pipenv run tests
+
+build_and_publish_docs:
+  tags:
+    - niamh
+  stage: test
+  script:
+    - pipenv run docs
+    - mkdir -p ${DOCS_PUBLISH_ROOT}/latest
+    - cp -rf build/html/* ${DOCS_PUBLISH_ROOT}/latest
```
Comment (thomascobb) [inline my_template.yml:32]: And we can put the print saying where the docs got published back in too...

File: my_template.yml (hunk 2 above)
Comment (niamhdougan) [reply my_template.yml:19]: As it's still a personal project on gitlab, I made my own runner.

File: .gitlab-ci.yml (hunk 3 above)
Comment (niamhdougan) [reply .gitlab-ci.yml:2]: Yep, working on that now.

File: my_template.yml (hunk 2 above)
Comment (niamhdougan) [reply my_template.yml:19]: Changing tag to rhel7 once added my_template.yml --> ci_templates/python3/dls_py3_template.yml

**Comment by aawdls** [comment]
Looks good, As discussed let's get the template added into `ci_templates` first and then include it from here. I'll look out for an MR on that project.

File: my_template.yml (hunk 1 above)
Comment (niamhdougan) [reply my_template.yml:23]: This will be changed to `pipenv run tests --cov-report html` in ci_templates

File: my_template.yml (hunk 4 above)
Comment (niamhdougan) [reply my_template.yml:32]: In merge request for ci_templates

**Comment by aawdls** [comment]
Do you know why Travis is unhappy?

**Comment by aawdls** [comment]
Oh those are the intentional failures to prompt you to fill in your own stuff, I see

**Pull request merged**